packages/types/src/codec/TypeRegistry.spec.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2017-2018 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import TypeRegistry from './TypeRegistry';
+import Text from '../Text';
+import U32 from '../U32';
+
+describe('TypeRegistry', () => {
+  let registry: TypeRegistry;
+
+  beforeEach(() => {
+    registry = new TypeRegistry();
+  });
+
+  it('Handles non exist type', () => {
+    expect(registry.get('non-exist')).toBeUndefined();
+  });
+
+  it('Able to register type', () => {
+    registry.register(Text);
+    expect(registry.get('Text')).toBe(Text);
+  });
+
+  it('Able to register type with a different name', () => {
+    registry.register('TextRenamed', Text);
+    expect(registry.get('TextRenamed')).toBe(Text);
```

Comment:
This is very cool. This would e.g. allow devs to choose BlockNumber as u128 or anything else.